Improve Colours on the Preview Tab's ROI Selectors

cailafinn opened the pull request:
**Description of work.**
Changed the colours and outlines of the ROI selectors on the Region Selector (SliceViewer) to make then more visible on a variety of colourmaps. 

**To test:**
1. Boot the Refl GUI
2. Switch to the preview tab
3. Load any file (45455 is best though)
4. Select a colourmap of choice on the region selector (middle widget)
5. Select a mix of signals, transmissions, and backgrounds
6. Check that it is clear which ROI is selected and what areas are covered by the various ROI selectors. 

Fixes #35349 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Commit:
Improve visibility of ROI selectors

RE #35349

Commit:
Add a release note

RE #35349

rbauststfc reviewed (commented):
I've tested adding the three different types of region against all the different available colour maps. It's not possible to get a set of region colours that are optimal for all, but the changes in this PR mean it's much easier to see the regions and to see when they are selected.

One suggestion, I noticed that the green for the signal region is not as easy to see against the colour for the data used in the default colour map (viridis). I also noticed that the blue for the transmission region is not that clear on a number of the colour maps (I think transmission regions would normally be selected as areas on the slice viewer without data points?). Would it be worth switching round the signal and transmission colours?

Commit:
Adjust colours and add hatching

Colours aren't a good way to differenciate between types of object, as
they're not likely to all be colourblind friendly. The hatching adds
this ability to differenciate without needing the colours.

RE #35349

Commit:
Fix tests for extra param

RE #35349

Commit:
Fix incorrect selector added in test

Accidentally replaced one too many of the test cases when adding the
hatch parameter.

RE #35349

rbauststfc reviewed (approved):
The hatching added to the regions is a great thought to distinguish the region type using something other than just colour and it worked well when I tested. This looks good to go in now.
